File: programmers/치킨 쿠폰.py
def solution(chicken):
    answer = 0
    while chicken // 10 != 0:
        a, b = divmod(chicken, 10)
        answer += a
        chicken = (a + b)

    return answer


print(solution(1999))
print(solution(1081))
print(solution(100))
print(solution(90))
print(solution(99))


print(solution(199))

File: programmers/코딩테스트 입문 with Java/치킨쿠폰.java
package 코딩테스트입문;

/**
 * https://school.programmers.co.kr/learn/courses/30/lessons/120884
 */
public class 치킨쿠폰 {
    public static void main(String[] args) {
        System.out.println(solution(100));
        System.out.println(solution(1081));
    }

    public static int solution(int chicken) {
        int answer = 0;
        while (true) {
            answer += (chicken / 10);
            chicken = chicken % 10 + chicken / 10;
            if (chicken < 10) {
                return answer;
            }
        }
    }
}
